packages/state-manager/src/sagas/publicChannels/publicChannels.selectors.ts

```diff
@@ -73,6 +73,28 @@ export const currentChannelAddress = createSelector(
   }
 )
 
+export const dynamicSearchedChannels = (channelInput: string) =>
+  createSelector(publicChannels, publicChannelsSelector => {
+    const recentChannels = publicChannelsSelector
+      .sort((a, b) => b.timestamp - a.timestamp)
+      .slice(0, 3)
+
+    const generalChannel = publicChannelsSelector.find(channel => channel.name === 'general')
+
+    const filteredList = publicChannelsSelector.filter(channel =>
+      channel.name.includes(channelInput)
+    )
+
+    const isFilteredList = filteredList.length > 0 ? filteredList : recentChannels
+
+    const areThreeRecentChannels = recentChannels.length >= 3 ? recentChannels : [generalChannel]
```

Comment:
You can include this condition in the selector I mentioned before. recentChannels selector could return either a list of channels or simply a general channel